## Final Project Submission

Please fill out:
* Student name: Grace Jepchumba Rotich
* Student pace:  full time
* Scheduled project review date/time: 
* Instructor name: Diana 
* Blog post URL: 


Imported necessary libraries and modules.

In [43]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns
import sqlite3

Using Panadas to read from the bom.movie_gross.csv file and check out the columns to understand the data.

In [44]:
bom_movie = pd.read_csv('Data/bom.movie_gross.csv')
bom_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This provides an overview of the DataFrame's structure, including the number of entries, columns, their names, data types, and memory usage. It's helpful for understanding the dataset's composition and characteristics at a glance.


Inorder to access the imdb file which is a sql database, a connection is created using the Sqlite3 module. Also utilizes the pd.read_sql() function from the pandas library to execute a SQL query against an SQLite database. 


In [45]:
conn = sqlite3.connect('./Data/im.db')
pd.read_sql("""
            SELECT name 
            FROM sqlite_master 
            WHERE type='table';
""", conn)

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

# **Data Presentation**



<img src="Images\datapresentation.png" alt="in image folder" style="width:800px;height:400px;">


In [46]:
bom_movie.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


The dataframe has 5 columns and 3387 rows of entries. There are some missing values in the studio, domestic gross and foreign gross columns.
Since the null values in the studio columns is few, lets see if we can drop them. First we filter the DataFrame bom_movie to include only those rows where the 'studio' column has missing values (NaN).

In [47]:
bom_movie.loc[bom_movie['studio'].isna() == True]

title studio  domestic_gross foreign_gross  year
210   Outside the Law (Hors-la-loi)    NaN         96900.0       3300000  2010
555         Fireflies in the Garden    NaN         70600.0       3300000  2011
933           Keith Lemon: The Film    NaN             NaN       4000000  2012
1862                 Plot for Peace    NaN          7100.0           NaN  2014
2825               Secret Superstar    NaN             NaN     122000000  2017

The above movies can be dropped since they are missing values and have made litte gross profits compaired to the rest of the movies in the data set.

In [48]:
bom_movie = bom_movie.dropna(subset=['studio'])

Lets checkout the top movies based off of their foreign_gross and domestic gross group them by the studio that produced them but foreign_gross is in string format so we should first convert to a float.We would strip all the commas from the strings in foreign gross and then convert them to float and replace the values with the new float values this will make it easier when using aggregate functions

In [49]:
bom_movie['foreign_gross'] = bom_movie['foreign_gross'].str.replace(',','')
bom_movie['foreign_gross'] = pd.to_numeric(bom_movie['foreign_gross'])

In [50]:
bom_movie.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   2033 non-null   float64
 4   year            3382 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 158.5+ KB


After fixing up the data type of the foreign gross column the next step would be to sort out the values to pick out the top ten movies based off of foreign gross that the movie earned and assigned it to a dataframe called top_10_movies_revenue

In [51]:
top_10_movies_revenue = bom_movie.sort_values(by=['foreign_gross','domestic_gross'], ascending=[False,False]).head(10)
top_10_movies_revenue

title studio  domestic_gross  \
328   Harry Potter and the Deathly Hallows Part 2     WB     381000000.0   
1875                      Avengers: Age of Ultron     BV     459000000.0   
727                         Marvel's The Avengers     BV     623400000.0   
3081               Jurassic World: Fallen Kingdom   Uni.     417700000.0   
1127                                       Frozen     BV     400700000.0   
2764                               Wolf Warrior 2     HC       2700000.0   
1477              Transformers: Age of Extinction   Par.     245400000.0   
1876                                      Minions   Uni.     336000000.0   
3083                                      Aquaman     WB     335100000.0   
1128                                   Iron Man 3     BV     409000000.0   

      foreign_gross  year  
328     960500000.0  2011  
1875    946400000.0  2015  
727     895500000.0  2012  
3081    891800000.0  2018  
1127    875700000.0  2013  
2764    867600000.0  2017  
1477    858600000.0  2014  
1876    823400000.0  2015  
3083    812700000.0  2018  
1128    805800000.0  2013

Created a dataframe called movie_info for the movie_info tsv file and observed the first 5 rows it contains to see if the data could be used for the project

In [52]:
movie_info = pd.read_csv('Data/rt.movie_info.tsv', delimiter='\t')
movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

Created a dataframe called tmovies for tmdb.movies.csv file and observed the first 5 rows it contains to see if the data could be used for the project

In [53]:
tmovies = pd.read_csv('Data/tmdb.movies.csv')
tmovies.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

Lets inspect the tmovies dataframe

In [54]:
tmovies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


There are 26517 entries and there are no missing values in the entire dataframe. Hence, w can use it without a lot of cleaning.
Sorting the dataframe using the vote_average , vote_count and popularity column.

In [55]:
tmovies = tmovies.sort_values(by=['vote_average','vote_count','popularity'], ascending=[False,False,False])
tmovies.head()

Unnamed: 0            genre_ids      id original_language  \
3687         3687  [35, 14, 18, 10749]  307124                en   
9199         9199  [35, 14, 18, 10749]  307124                en   
19174       19174              [10402]  411990                en   
12742       12742                 [99]  276913                en   
24326       24326             [16, 12]  561974                ko   

                             original_title  popularity release_date  \
3687                        Love on a Leash       1.400   2013-08-13   
9199                        Love on a Leash       1.400   2013-08-13   
19174                               Endless       1.003   2016-08-19   
12742  Hidden Colors 3: The Rules of Racism       0.968   2014-06-26   
24326                 점박이 한반도의 공룡 2: 새로운 낙원       7.941   2018-12-31   

                                        title  vote_average  vote_count  
3687                          Love on a Leash          10.0           7  
9199                          Love on a Leash          10.0           7  
19174                                 Endless          10.0           4  
12742    Hidden Colors 3: The Rules of Racism          10.0           4  
24326  Dino King 3D: Journey to Fire Mountain          10.0           3

AS shown above, there are some duplicates to be dropped for better data analysis. Also dropped the unnamed and genre_ids columns since its data cannot be understood.

In [56]:
#dropping the unnamed and genre_id columns
tmovies = tmovies.drop(['Unnamed: 0','genre_ids'], axis=1)
#dropping any duplicated
tmovies.drop_duplicates(inplace=True)
tmovies.head()



id original_language                        original_title  \
3687   307124                en                       Love on a Leash   
19174  411990                en                               Endless   
12742  276913                en  Hidden Colors 3: The Rules of Racism   
24326  561974                ko                 점박이 한반도의 공룡 2: 새로운 낙원   
24612  541576                en                     Meeting Gorbachev   

       popularity release_date                                   title  \
3687        1.400   2013-08-13                         Love on a Leash   
19174       1.003   2016-08-19                                 Endless   
12742       0.968   2014-06-26    Hidden Colors 3: The Rules of Racism   
24326       7.941   2018-12-31  Dino King 3D: Journey to Fire Mountain   
24612       5.870   2019-05-03                       Meeting Gorbachev   

       vote_average  vote_count  
3687           10.0           7  
19174          10.0           4  
12742          10.0           4  
24326          10.0           3  
24612          10.0           3

Created a dataframe called movie_budgets and opened the movie_budgets.csv file and looked at the info of the data entries and the columns

In [57]:
movie_budgets = pd.read_csv('./Data/tn.movie_budgets.csv')
movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


The movie_budgets is a full dataset with no null values thus can be worked with without much cleaning to do.An excellent dataset.Looked at the first 5 rows of the dataset to get a feel of the data

In [58]:
movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

The production_budget,domestic_gross and worldwide gross are all in string format and they are numbers that could be used also theres commas in between 3 digits so that would have to be stripped off together with the dollar sign($) and converted to floats to be used .This will be done in the cell below

In [59]:
movie_budgets['production_budget'] = movie_budgets['production_budget'].str.replace(',', '').str.replace('$','').astype('float')
movie_budgets['domestic_gross'] = movie_budgets['domestic_gross'].str.replace(',', '').str.replace('$','').astype('float')
movie_budgets['worldwide_gross'] = movie_budgets['worldwide_gross'].str.replace(',', '').str.replace('$','').astype('float')
movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5782 non-null   int64  
 1   release_date       5782 non-null   object 
 2   movie              5782 non-null   object 
 3   production_budget  5782 non-null   float64
 4   domestic_gross     5782 non-null   float64
 5   worldwide_gross    5782 non-null   float64
dtypes: float64(3), int64(1), object(2)
memory usage: 271.2+ KB


Arranged the movie_budgets based off of the most earned worldwide_gross,domestic_gross and production_budget

In [60]:
movie_budgets = movie_budgets.sort_values(by=['worldwide_gross','domestic_gross','production_budget'],ascending=[False,False,True])
movie_budgets.head()

id  release_date                                 movie  production_budget  \
0    1  Dec 18, 2009                                Avatar        425000000.0   
42  43  Dec 19, 1997                               Titanic        200000000.0   
5    6  Dec 18, 2015  Star Wars Ep. VII: The Force Awakens        306000000.0   
6    7  Apr 27, 2018                Avengers: Infinity War        300000000.0   
33  34  Jun 12, 2015                        Jurassic World        215000000.0   

    domestic_gross  worldwide_gross  
0      760507625.0     2.776345e+09  
42     659363944.0     2.208208e+09  
5      936662225.0     2.053311e+09  
6      678815482.0     2.048134e+09  
33     652270625.0     1.648855e+09

That a lot of money!
Nice now to the im.db file created a function called readsql where the argument would be an sql query and it will return a result

In [61]:
def readsql(query):
    return pd.read_sql(query, conn)

Created a dataframe called Imdb list that joins the movie_ratings and movie basics using movie_id

In [64]:
imdblist = readsql("""SELECT *
FROM movie_basics
JOIN movie_ratings 
USING(movie_id)
""")
imdblist.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   movie_id         73856 non-null  object 
 1   primary_title    73856 non-null  object 
 2   original_title   73856 non-null  object 
 3   start_year       73856 non-null  int64  
 4   runtime_minutes  66236 non-null  float64
 5   genres           73052 non-null  object 
 6   averagerating    73856 non-null  float64
 7   numvotes         73856 non-null  int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 4.5+ MB


In [63]:
imdblist.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  averagerating  numvotes  
0        2013            175.0    Action,Crime,Drama            7.0        77  
1        2019            114.0       Biography,Drama            7.2        43  
2        2018            122.0                 Drama            6.9      4517  
3        2018              NaN          Comedy,Drama            6.1        13  
4        2017             80.0  Comedy,Drama,Fantasy            6.5       119

Since the start year is in integer format and its just years and for the movie_budget dataframe has the date month and year creating another column called year would be ideal for merging the two data frames so that we dont have two movies with a commonn name mixed together

In [65]:
movie_budgets['year'] = movie_budgets['release_date'].astype(str).str[-4:]    
movie_budgets['year'] = movie_budgets['year'].astype('int64')
imdb_budget = pd.merge(movie_budgets,imdblist,left_on=['movie', 'year'],right_on=['primary_title','start_year'])
imdb_budget.head()

id  release_date                    movie  production_budget  \
0   7  Apr 27, 2018   Avengers: Infinity War        300000000.0   
1  34  Jun 12, 2015           Jurassic World        215000000.0   
2  67   Apr 3, 2015                Furious 7        190000000.0   
3  27   May 4, 2012             The Avengers        225000000.0   
4   4   May 1, 2015  Avengers: Age of Ultron        330600000.0   

   domestic_gross  worldwide_gross  year   movie_id            primary_title  \
0     678815482.0     2.048134e+09  2018  tt4154756   Avengers: Infinity War   
1     652270625.0     1.648855e+09  2015  tt0369610           Jurassic World   
2     353007020.0     1.518723e+09  2015  tt2820852                Furious 7   
3     623279547.0     1.517936e+09  2012  tt0848228             The Avengers   
4     459005868.0     1.403014e+09  2015  tt2395427  Avengers: Age of Ultron   

            original_title  start_year  runtime_minutes  \
0   Avengers: Infinity War        2018            149.0   
1           Jurassic World        2015            124.0   
2            Furious Seven        2015            137.0   
3             The Avengers        2012            143.0   
4  Avengers: Age of Ultron        2015            141.0   

                    genres  averagerating  numvotes  
0  Action,Adventure,Sci-Fi            8.5    670926  
1  Action,Adventure,Sci-Fi            7.0    539338  
2    Action,Crime,Thriller            7.2    335074  
3  Action,Adventure,Sci-Fi            8.1   1183655  
4  Action,Adventure,Sci-Fi            7.3    665594